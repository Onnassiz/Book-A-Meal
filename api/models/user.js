
module.exports = (sequelize, DataTypes) => {
	const User = sequelize.define('user', {
		id: {
			allowNull: false,
			primaryKey: true,
			type: DataTypes.UUID,
			defaultValue: DataTypes.UUIDV4,
		},
		email: {
			type: DataTypes.STRING,
			validate: {
				isEmail: true,
			},
		},
		passwordHash: DataTypes.STRING,
		fullName: DataTypes.STRING,
		role: DataTypes.STRING,
	}, {});
	User.associate = (models) => {
		User.hasMany(models.meal);
		User.hasMany(models.menu);
		User.hasOne(models.profile);
	};
	return User;
};
