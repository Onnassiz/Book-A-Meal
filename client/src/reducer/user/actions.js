import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actions';

export const SET_USER = 'SET_USER';
export const SET_SIGN_UP_SERVER_ERRORS = 'SET_SIGN_UP_SERVER_ERRORS';
export const SET_SIGN_IN_SERVER_ERRORS = 'SET_SIGN_IN_SERVER_ERRORS';

export function setUser(user) {
  return dispatch => dispatch({
    type: SET_USER,
    id: user.id,
    name: user.name,
    role: user.role,
    email: user.email,
  });
}

function setSignUpErrors(errors) {
  return dispatch => dispatch({
    type: SET_SIGN_UP_SERVER_ERRORS,
    errors,
  });
}

function setSignInErrors(errors) {
  return dispatch => dispatch({
    type: SET_SIGN_IN_SERVER_ERRORS,
    errors,
  });
}

export function postUser(user) {
  return (dispatch) => {
    dispatch(setLoading());
    axios.post('api/v1/auth/signUp', user).then((response) => {
      localStorage.setItem('id', response.data.id);
      localStorage.setItem('email', response.data.email);
      localStorage.setItem('name', response.data.fullName);
      localStorage.setItem('role', response.data.role);
      localStorage.setItem('token', response.data.token);
      setTimeout(() => {
        window.location = '/';
      }, 500);
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setSignUpErrors(error.response.data));
    });
  };
}


export function signInUser(user) {
  return (dispatch) => {
    dispatch(setLoading());
    axios.post('api/v1/auth/signIn', user).then((response) => {
      localStorage.setItem('id', response.data.id);
      localStorage.setItem('email', response.data.email);
      localStorage.setItem('name', response.data.fullName);
      localStorage.setItem('role', response.data.role);
      localStorage.setItem('token', response.data.token);
      setTimeout(() => {
        window.location = '/';
      }, 500);
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setSignInErrors(error.response.data));
    });
  };
}
