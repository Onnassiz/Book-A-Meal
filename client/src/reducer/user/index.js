import { SET_USER, SET_SIGN_UP_SERVER_ERRORS, SET_SIGN_IN_SERVER_ERRORS } from './action';

const initialState = {};

export default (state = initialState, action) => {
	switch (action.type) {
	case SET_USER:
		return { name: action.name, email: action.email, role: action.role };
	case SET_SIGN_UP_SERVER_ERRORS:
		return { signUpErrors: action.errors };
	case SET_SIGN_IN_SERVER_ERRORS:
		return { signInErrors: action.errors };
	default:
		return state;
	}
};
