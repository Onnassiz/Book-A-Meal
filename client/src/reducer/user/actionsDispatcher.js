import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actionsDispatcher';
import { setUserAction, setSignUpErrorsAction, setSignInErrorsAction } from './actions';

export const setUser = user => dispatch => dispatch(setUserAction(user));

const setSignUpErrors = errors => dispatch => dispatch(setSignUpErrorsAction(errors));
const setSignInErrors = errors => dispatch => dispatch(setSignInErrorsAction(errors));

const processSignIn = (data) => {
  localStorage.setItem('id', data.id);
  localStorage.setItem('email', data.email);
  localStorage.setItem('name', data.fullName);
  localStorage.setItem('role', data.role);
  localStorage.setItem('token', data.token);
  setTimeout(() => {
    window.location = '/';
  }, 500);
};

export const postUser = user => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/auth/signUp', user).then((response) => {
    dispatch(unsetLoading());
    processSignIn(response.data);
  }).catch((error) => {
    dispatch(setSignUpErrors(error.response.data));
    dispatch(unsetLoading());
  });
};


export const signInUser = user => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/auth/signIn', user).then((response) => {
    dispatch(unsetLoading());
    processSignIn(response.data);
  }).catch((error) => {
    dispatch(setSignInErrors(error.response.data));
    dispatch(unsetLoading());
  });
};
