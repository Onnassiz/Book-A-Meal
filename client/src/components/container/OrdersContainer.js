import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Orders from '../presentation/Orders';
import { getUserOrders, getMealsInOrder, setUpdatedOrder } from '../../reducer/orders/actionsDispatcher';
import { addArrayToCart } from '../../reducer/cart/actionsDispatcher';


const mapStateToProps = state => ({
  orders: state.orders,
  user: state.user,
  formState: state.formState,
});


function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    getUserOrders,
    getMealsInOrder,
    setUpdatedOrder,
    addArrayToCart,
  }, dispatch);
}


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Orders);
