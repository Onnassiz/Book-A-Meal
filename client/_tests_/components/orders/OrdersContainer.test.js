import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import user from '../../objectProps/user.props';
import OrdersContainer from '../../../src/components/container/OrdersContainer';

const mockStore = configureStore();

const store = mockStore({
  orders: {},
  formState: { isLoading: false },
  user,
});

const wrapper = shallow(<OrdersContainer store={store} />);

describe('<OrdersContainer />', () => {
  it('Should render the OrdersContainer container successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
