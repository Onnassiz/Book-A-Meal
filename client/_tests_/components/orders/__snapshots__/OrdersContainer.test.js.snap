// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersContainer /> Should render the OrdersContainer container successfully 1`] = `
<Orders
  addArrayToCart={[Function]}
  formState={
    Object {
      "isLoading": false,
    }
  }
  getMealsInOrder={[Function]}
  getUserOrders={[Function]}
  orders={Object {}}
  setUpdatedOrder={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "onnassiz@gmail",
      "id": "3a3c499f-803a-4e06-a328-307daff22243",
      "name": "Ben Onah",
      "role": "caterer",
    }
  }
/>
`;
